src/app/core/core.module.ts

```diff
@@ -0,0 +1,14 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { PaginationModule } from 'ngx-bootstrap/pagination';
+
+import { HttpClientModule } from '@angular/common/http';
+import { ProductService } from './ProductService';
+import { ContactService } from './ContactService';
+
+@NgModule({
+  declarations: [],
+  imports: [CommonModule, HttpClientModule, PaginationModule.forRoot()],
+  providers: [ProductService, ContactService],
```

Comment:
jesli serwisy maja w dekoratorze providedIn: root to to jest niepotrzebne. Wystarczy je zgrupowac w katalogu z core modulem i albo zostawic im providedIn:root  albo rejestrowanie w providers w module :)

mam tez watpliwosci co do productSerwis - czy nie powinien byc on wydzielony do modulu z produktami?

Reply:
Zrobilem to po komentarzach na CR i naszej rozmowie: bedzie podzial na moduly, serwisy beda w modulach, serwisy same w sobie beda mialy tylko dekorator @Injectable() bez ProvidedIn a tutaj zostawie deklaracje